shell/gpu/gpu_surface_metal_delegate.h

```diff
@@ -0,0 +1,104 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_GPU_GPU_SURFACE_METAL_DELEGATE_H_
+#define FLUTTER_SHELL_GPU_GPU_SURFACE_METAL_DELEGATE_H_
+
+#include <stdint.h>
+
+#include "flutter/fml/macros.h"
+#include "third_party/skia/include/core/SkSurface.h"
+#include "third_party/skia/include/gpu/mtl/GrMtlTypes.h"
+
+namespace flutter {
+
+struct MTLFrameInfo {
+  uint32_t width;
+  uint32_t height;
+};
+
+// expected to be id<MTLDevice>
+typedef void* GPUMTLDeviceHandle;
+
+// expected to be id<MTLCommandQueues>
+typedef void* GPUMTLCommandQueueHandle;
+
+// expected to be CAMetalLayer*
+typedef void* GPUCAMetalLayerHandle;
```

Comment:
Can these be `fml::scoped_nsobject`? Or are you including this header in a non-ObjC TU?

Reply:
As discussed, this is used from a non-objC TU.